paccmann_chemistry/models/training.py

```diff
@@ -123,14 +123,18 @@ def train_vae(
     t = time()
     for _iter, batch in enumerate(train_dataloader):
         global_step = (epoch - 1) * len(train_dataloader) + _iter
-        padded_batch = torch.nn.utils.rnn.pad_sequence(batch)
-        padded_batch = padded_batch.to(device)
-        encoder_seq, decoder_seq, target_seq = sequential_data_preparation(
-            padded_batch,
-            input_keep=input_keep,
-            start_index=start_index,
-            end_index=end_index
-        )
+        # padded_batch = torch.nn.utils.rnn.pad_sequence(batch)
+        # packed_batck = torch.nn.utils.rnn.pack_sequence(batch)
+        # padded_batch = padded_batch.to(device)
+
+        # WIP: We will use packed sequences instead of padded sequences
+        (encoder_seq, decoder_seq,
+         target_seq) = packed_sequential_data_preparation(
```

Comment:
@jannisborn I need your opinion. Do you think it makes sense to put this inside the GRU?

I am working on the bidirectional part (i.e. the backward GRU) and if I want to keep using this data preparation for outside the model I need to unpack the sequence, reverse it, and re-pack it; which is a bit cumbersome. At the same time, if we want to use the model we need to always call this `sequential data perparation` step, so wouldn't it be more convenient if it was incorporated into the model?

Reply:
Hm, I think we shouldn't mix generic data utils with models and would favor a solution that outsources `packed_sequential_data_preparation `. 
On the second part, in #6 we agreed to not flip brackets for the backward GRU and if possible, I would do the same here, just feeding the sequence backwards, starting from the StopToken. This may sound weird, but it shouldn't matter as the backward GRU is decoupled from the forward and in the end, it's just about input statistics. Hence, I would refrain from any additional operations in the packing for bidirectional GRUl, a bit similar to [this](https://stackoverflow.com/questions/49203019/how-to-use-pack-padded-sequence-with-multiple-variable-length-input-with-the-sam).

Reply:
Not sure it answers all the questions, ping me if not!

Reply:
Okay, it makes sense. I will then add a check at the beggining of the call to see if the sequence is an instance of `PackedSequence` and (if so) run the logic I am creating here. Regarding to the link you mentioned, I will check it. I already pushed a draft for the bidirectional part, that I think kinda works (for now at least), so I'll compare that to the stackoverflow post.

Reply:
Oh, and yeah, I won't invert the tokens (I'm already doing that in the last commit). If it's an independent model it makes sorta sense to do it like this.